tracer/src/Datadog.Trace/Debugger/Symbols/SymbolsUploader.cs

```diff
@@ -0,0 +1,289 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Debugger.Sink;
+using Datadog.Trace.Debugger.Symbols.Model;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.Debugger.Symbols
+{
+    internal class SymbolsUploader : ISymbolsUploader
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SymbolsUploader));
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private readonly string _serviceName;
+        private readonly string? _serviceVersion;
+        private readonly string? _environment;
+        private readonly SemaphoreSlim _assemblySemaphore;
+        private readonly SemaphoreSlim _discoveryServiceSemaphore;
+        private readonly HashSet<string> _alreadyProcessed;
+        private readonly long _thresholdInBytes;
+        private readonly CancellationTokenSource _cancellationToken;
+        private readonly IBatchUploadApi _api;
+        private IDiscoveryService? _discoveryService;
+        private byte[]? _payload;
+        private string? _isSymbolUploaderEnabled;
+
+        private SymbolsUploader(IBatchUploadApi api, IDiscoveryService discoveryService, DebuggerSettings settings, ImmutableTracerSettings tracerSettings, string serviceName)
+        {
+            _isSymbolUploaderEnabled = null;
+            _alreadyProcessed = new HashSet<string>();
+            _environment = tracerSettings.EnvironmentInternal;
+            _serviceVersion = tracerSettings.ServiceVersionInternal;
+            _serviceName = serviceName;
+            _discoveryService = discoveryService;
+            _api = api;
+            _assemblySemaphore = new SemaphoreSlim(1);
+            _discoveryServiceSemaphore = new SemaphoreSlim(0);
+            _thresholdInBytes = settings.SymbolBatchSizeInBytes;
+            _cancellationToken = new CancellationTokenSource();
+            _jsonSerializerSettings = new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore };
+            _discoveryService.SubscribeToChanges(ConfigurationChanged);
+        }
+
+        private void ConfigurationChanged(AgentConfiguration configuration)
+        {
+            _isSymbolUploaderEnabled = configuration.SymbolDbEndpoint;
+            if (string.IsNullOrEmpty(Volatile.Read(ref _isSymbolUploaderEnabled)))
+            {
+                Log.Warning("Dynamic Instrumentation failed to upload symbols for auto-complete suggestions. Ensure that you are working with datadog-agent v7.45.0 or higher");
```

Comment:
Because Dynamic Instrumentation does support v7.45 right now according to the docs, and until we decide to change that, it seems wrong to give them a warning as if they did something wrong or make them worry that something 'failed'. I'd suggest either removing this warning or making it less 'scary'.

Reply:
Fixed in db6086b623f7ad8ae43181865be84140bce1ef3c

Reply:
nit: should be 'lower than' instead of 'less then'

Reply:
Fixed in d76bac0e9bd3cfbda3e480deff9bbd0e453a32ec